src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,121 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+/**
+ * The `Event` class represents a task that spans a specific duration in Duke.
+ */
+public class Event extends Task {
+
+    /**
+     * The formatted string representation of the start time.
+     */
+    protected String from;
+
+    /**
+     * The formatted string representation of the end time.
+     */
+    protected String to;
+
+    /**
+     * The input date-time formatter.
+     */
+    protected DateTimeFormatter input = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * The output date-time formatter for display.
+     */
+    protected DateTimeFormatter output = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+
+    /**
+     * Constructs an `Event` with the given description, start time, and end time.
+     *
+     * @param description The description of the event.
+     * @param from The start time of the event in yyyy-MM-dd HH:mm format.
+     * @param to The end time of the event in yyyy-MM-dd HH:mm format.
+     * @throws DukeException If there is an issue parsing the time or the format is invalid.
+     */
+    public Event(String description, String from, String to) throws DukeException {
+        super(description);
+        try {
+            LocalDateTime date1 = LocalDateTime.parse(from, input);
+            LocalDateTime end1 = LocalDateTime.parse(to, input);
+            this.from = date1.format(output);
+            this.to = end1.format(output);
+
+            if (!isBeforeDateTime(date1, end1)) {
+                throw new DukeException("Invalid date range. 'From' should be before 'To'.");
+            }
+        } catch (DateTimeParseException | StringIndexOutOfBoundsException e) {
+            throw new DukeException("Invalid time format. Please use yyyy-MM-dd HH:mm.");
+        }
+    }
```

Comment:
I like how clean this code is 👍 Also great that you catch errors when the dates are not in the correct order!